src/core/stream/adaptation/adaptation_stream.ts

```diff
@@ -375,6 +377,9 @@ export default function AdaptationStream({
       const bufferGoal$ = wantedBufferAhead.asObservable().pipe(
         map((wba) => wba * bufferGoalRatio)
       );
+      const maxVideoBufferSize$ = maxVideoBufferSize.asObservable();
+      const maxBufferSize$ = adaptation.type === "video" ? maxVideoBufferSize$ :
+                                                           observableOf(Infinity);
```

Comment:
I would prefer only calling `asObservable` when we need it, just for principles (we can do that by including the call in the ternary's first case).

Reply:
Yes, I was wary as it exceed the limited line length